### PR TITLE
Add pw reset to salt login and make link always visible on bountysource

### PR DESCRIPTION
**Pull request by mitnuh**

### CODE REVIEW

**Comment by mitnuh** [comment]
The first commit added the reset link too salt but relied on the existing reset password form on bountysource. 

The latest commit adds a password reset form to salt and tweaks the reset password email so the reset link is to whichever website the request was on.

It works like bountysource reset form but looks a bit more like the salt login form.

File: app/controllers/api/v1/people_controller.rb [hunk 1]
```diff
@@ -247,7 +247,13 @@ def request_password_reset
     require_params(:email)
 
     if (person = Person.find_by_email(params[:email]))
-      person.send_email(:reset_password)
+      if URI(request.referer).host == URI.parse(Api::Application.config.salt_url).host
```
Comment (wkonkel) [inline app/controllers/api/v1/people_controller.rb]: I don't think `request.referer` is reliable... could we instead pass a `is_salt=true` boolean as a parameter?

Comment (mitnuh) [inline app/controllers/api/v1/people_controller.rb]: Will do! Was hoping to do this today but tomorrow is more likely.

**Comment by mitnuh** [comment]
Changed to use is_salt parameter and added a test for the people_controller. 

If you have suggestions or advice for more/better tests they'd be appreciated! I considered adding a test to mailer_spec.rb for the extra parameter added to the email template but wasn't sure it would be a useful test...

**Comment by mitnuh** [comment]
I need to git rebase because these commits are on my master branch and they're cluttering an unrelated PR. Might need to close this and make a new PR.

**Pull request closed without merging**